cpp/src/arrow/util/async_generator.h

```diff
@@ -800,16 +816,18 @@ class ReadaheadGenerator {
         : source_generator(std::move(source_generator)), max_readahead(max_readahead) {}
 
     void MarkFinishedIfDone(const T& next_result) {
+      // ASSERT_HELD(mu)
       if (IsIterationEnd(next_result)) {
-        finished.store(true);
+        finished = true;
       }
     }
 
     AsyncGenerator<T> source_generator;
     int max_readahead;
     Future<> final_future = Future<>::Make();
-    std::atomic<int> num_running{0};
-    std::atomic<bool> finished{false};
+    int num_running{0};    // GUARDED_BY(mu)
```

Comment:
Is `GUARDED_BY` an annotation for some kind of static analysis tool? Which one?

Reply:
This is just my code style, related code is in https://clang.llvm.org/docs/ThreadSafetyAnalysis.html